app/models/electronic_pharmacy_stock_record.rb

```diff
@@ -19,8 +20,8 @@ def error?
     obfuscation_state == 'error'
   end
 
-  def rows
-    JSON[self[:rows]]
+  def read_rows
```

Comment:
Is this method in use? Try using `.open` instead of `.download` to avoid loading the blob into memory.

Reply:
Nope. Actually, it wasn't being used so I deleted it